### DISCUSSION TITLE
Xamaring Forms compatibility.

### DISCUSSION THREAD

**Discussion by jazzzsoftware** [dlemstra/Magick.NET#655] [Help]
Hi,

Can this library be used with Xamarin Forms (.Net Satandard 2.0)? 

I created a new XF project and added Nuget package Magick.NET-Q8-AnyCPU 7.19.0 to all projects. 

I have this code:

`using (MagickImage image = new MagickImage(stream))`

On iOS i get the following error:
System.TypeInitializationException: 'The type initializer for 'NativeMagickSettings' threw an exception.'

and Android:
System.DllNotFoundException: 'Magick.Native-Q8-x86.dll assembly:<unknown assembly> type:<unknown type> member:(null)'

How can I fix this?

Thanks

Mark.

PS. Can provide a sample project if it would be helpful.

**Comment by dlemstra** [comment]
You should be able to use the library in that project but it seems that you manually need to copy the native library from the runtimes folder of the NuGet package. I might be able to automate that if you provide me with a sample project that I can open in VisualStudio on Windows.

**Comment by jazzzsoftware** [comment]
Awesome thank you. Here is a standard XF project with Magick.Net added.
[MagickDotNet.zip](https://github.com/dlemstra/Magick.NET/files/4744416/MagickDotNet.zip)

I will try your manual copy suggestion.

**Reply by dlemstra** [reply]
It looks like a `project.assets.json` is generated in the `obj` folder for the native library and that contains a reference to the native library. Not sure why this is not copied to the bin folder. The trick that is done for `net20` and `net40` won't work in this situation.
